src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -416,31 +458,65 @@ private function evaluateScalar(string $expr, mixed $context): mixed
         return null;
     }
 
-    private function evaluateFunction(string $name, string $args, array $context): mixed
+    private function evaluateFunction(string $name, string $args, mixed $context): mixed
     {
-        $args = array_map(
-            fn ($arg) => $this->evaluateScalar(trim($arg), $context),
-            explode(',', $args)
-        );
+        $argList = [];
+        $nodelistSizes = [];
+        if ($args = trim($args)) {
+            $args = $this->parseCommaSeparatedValues($args);
+            foreach ($args as $arg) {
+                $arg = trim($arg);
+                if (str_starts_with($arg, '@')) { // special handling for @ to track nodelist size
```

Comment:
this will reduce the nesting level:
```suggestion
                if (str_starts_with($arg, '$')) { // special handling for absolute paths
                }elseif (!str_starts_with($arg, '@')) {
                //[...]
```

Reply:
Ah yes good catch!